Add ability to add works to existing list from Bulk Search

Shikhar-395 opened the pull request:
closes #11348 

### **Feature Requested**

Currently the bulk search endpoint is a handy UI to easily support adding many books to a new list. Extending this to add books to an already existing list would be a useful expansion.

**### Solution**
Backend:
Reuse the logic behind /lists/add, just extending it to handle adding works to an existing list by ID.

Frontend:
When the user clicks “Add to List” on the bulk search page, a dropdown 
will open showing their existing lists. They can either pick one to add the selected works to or click “Create new list” to go through the current /lists/add flow. 

https://github.com/user-attachments/assets/9f149f10-6863-4284-aa6c-a5bb4117f06a

**Stakeholders** @cdrini

cdrini commented on openlibrary/components/BulkSearch/components/BulkSearchControls.vue:
```diff
@@ -32,15 +32,41 @@
-                :key="index"
-                :value="index"
+            <label><strong> Extractor</strong> <br>
+              <div
```
We want to use the bulit-in `<select>` element here ; that has better baseline accessibility features, and supports the builtin vue attributes for two-way data-binding, `v-model`. Otherwise we have to do all that manually!

cdrini commented on openlibrary/components/BulkSearch/components/BulkSearchControls.vue:
```diff
@@ -179,15 +263,107 @@ export default {
+        this.loadUserLists()
+        document.addEventListener('click', this.closeDropdown)
+        window.addEventListener('scroll', this.closeDropdown, true)
+        window.addEventListener('resize', this.closeDropdown)
```
This is not standard, and can cause accessibility issues.

Shikhar-395 committed:
feat: Add ability to add works to existing list from Bulk Search (#11348)

- Added dropdown to select between creating new list or adding to existing list
- Replaced native select elements with custom dropdowns for extractor and list selection
- Disabled 'Save to' dropdown interaction based on createListDisabled status
- Added activeStep watcher for smooth scrolling to active progress step

Shikhar-395 committed:
Remove active step highlighting from bulk search progress cards and an instruction paragraph.

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

Shikhar-395 committed:
feat: enable adding seeds to a list via URL parameter on the edit page and update list add link to edit.

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci
